### PR TITLE
add data augmentation to sanakoyeu_et_al_2018

### PR DESCRIPTION
**Pull request by pmeier**
This adds the image augmentation pipeline that was split off from #162 (and later #177 ). For the augmentation transforms we pull `kornia` as dependency. Given that `pystiche` has already started to replace the custom transforms with `kornia` and it will be a dependency in the future anyway, this should be fine. 

ToDo:

- [x] Tests

### CODE REVIEW

File: pystiche_papers/sanakoyeu_et_al_2018/_augmentation.py [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+    high: Union[float, torch.Tensor],
+    same_on_batch: bool = False,
+) -> torch.Tensor:
+    return _adapted_uniform(shape, low, high + 1 - 1e-6, same_on_batch).int()
+
+
+class RandomCrop(AugmentationBase):
+    def __init__(
+        self,
+        size: Union[Tuple[int, int], int],
```
Comment (jbueltemeier) [inline pystiche_papers/sanakoyeu_et_al_2018/_augmentation.py:217]: The authors use a default `crop_size` of `(256,256)`. See [here](https://github.com/CompVis/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/img_augm.py#L26).

File: pystiche_papers/sanakoyeu_et_al_2018/_data.py [hunk 2]
```diff
@@ -84,22 +85,25 @@ def _properties(self) -> Dict[str, Any]:
...
-) -> transforms.ComposedTransform:
-    transform = style_image_transform(edge_size)
+    impl_params: bool = True, edge_size: int = 768, train: bool = False,
+) -> nn.Sequential:
+    transform = style_image_transform(edge_size=edge_size, train=train)
     if not impl_params:
         return transform
 
-    return transforms.ComposedTransform(transforms.Rescale(2.0), transform)
+    return nn.Sequential(transforms.Rescale(2.0), *transform.children())
```
Comment (jbueltemeier) [inline pystiche_papers/sanakoyeu_et_al_2018/_data.py:106]: During `stylization`, only the smaller side of the image is resized to the `image_size`. See [here](https://github.com/CompVis/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/model.py#L492-L494).

File: pystiche_papers/sanakoyeu_et_al_2018/_augmentation.py (hunk 1 above)
Comment (pmeier) [reply pystiche_papers/sanakoyeu_et_al_2018/_augmentation.py:217]: That is true, but they are [not using the default](https://github.com/CompVis/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/model.py#L271) during training.

File: pystiche_papers/sanakoyeu_et_al_2018/_data.py (hunk 2 above)
Comment (pmeier) [reply pystiche_papers/sanakoyeu_et_al_2018/_data.py:106]: This refers to https://github.com/CompVis/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/prepare_dataset.py#L133. We should have done this in #177. I'll send a follow-up PR.

**Review by jbueltemeier — APPROVED** [review]

**Pull request merged**